src/System.Management.Automation/FormatAndOutput/common/Utilities/Mshexpression.cs

```diff
@@ -246,25 +301,36 @@ private MshExpressionResult GetValue(PSObject target, bool eatExceptions)
                     PSMemberInfo member = target.Properties[_stringValue];
                     if (member == null)
                     {
-                        return new MshExpressionResult(null, this, null);
+                        return new PSPropertyExpressionResult(null, this, null);
                     }
                     result = member.Value;
                 }
 
-                return new MshExpressionResult(result, this, null);
+                return new PSPropertyExpressionResult(result, this, null);
             }
             catch (RuntimeException e)
             {
                 if (eatExceptions)
                 {
-                    return new MshExpressionResult(null, this, e);
+                    return new PSPropertyExpressionResult(null, this, e);
                 }
                 else
                 {
                     throw;
                 }
             }
         }
+        
+        private PSObject IfHashtableWrapAsPSCustomObject(PSObject target)
+        {
+            // If the object passed in is a hashtable, then turn it into a PSCustomObject so
+            // that property expressions can work on it.
+            if (PSObject.Base(target) is Hashtable targetAsHash)
```

Comment:
If we want to handle hashtable specially, shall we expand that to `IDictionary`? `LanguagePrimitives.ConvertPSObjectToType` is able to convert `IDictionary` to `PSObject`.

Reply:
The goal is to match the behaviour of `[PSCustomObject] @{a=1; b=2}`. Currently casting `Dictionary<T,T>` to `PSCustomObject` doesn't result in  a custom object so I don't think we should introduce that behaviour here.

Reply:
Understood. Good to know that we are matching the behavior of `[PSCustomObject] @{a=1; b=2}`.